File: C/exercice4_32.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>

int main(int argc, char const *argv[])
{
	int fd ;
	FILE *fp ;
	char chaine[50] ;
	fd = open(argv[1], O_RDONLY) ;
	fp = fdopen(fd, "r") ;
	fscanf(fp, "%s\n", chaine) ;
	puts(chaine) ;
	unlink(argv[1]) ;
	return 0;
}

File: Python/exercice4_32.py
import os
import sys
import time

def main():
    if len(sys.argv) != 2:
        print("Utilisation: python script.py <nom_fichier>")
        return 1
    
    nom_fichier = sys.argv[1]
    
    with open(nom_fichier, 'r') as f:
        chaine = f.readline().strip()
        print(chaine)
            
    os.unlink(nom_fichier)

if __name__ == "__main__":
    main()